src/client/components/sms-fraud/SubmitCodeForm.tsx

```diff
@@ -0,0 +1,107 @@
+import { SendMessageButton } from './SendSMSMessageButton';
+import { Alert } from '../common/Alert/Alert';
+import Button from '../common/Button/Button';
+import styles from './smsVerificationFraud.module.scss';
+import formStyles from '../../../styles/forms.module.scss';
+import { SendMessageMutation } from '../../../pages/sms-fraud';
+
+export const useSubmitCode = (params?: { onSuccess?: () => void }) => {
+  const { getData } = useVisitorData(
+    { ignoreCache: true },
+    {
+      immediate: false,
+    },
+  );
+  return useMutation<SubmitCodeResponse, Error, { phoneNumber: string; code: string }>({
+    mutationKey: ['submitCode'],
+    mutationFn: async ({ code, phoneNumber }) => {
+      const { requestId } = await getData();
+      const response = await fetch(`/api/sms-fraud/submit-code`, {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          code: Number(code),
+          phoneNumber,
+          requestId,
+        } satisfies SubmitCodePayload),
+      });
+      if (response.status < 500) {
+        return await response.json();
+      } else {
+        throw new Error('Failed to submit code: ' + response.statusText);
+      }
+    },
+    onSuccess: (data) => {
+      if (data.severity === 'success') {
+        params?.onSuccess?.();
+      }
+    },
+  });
+};
+
+type SubmitCodeFormProps = {
+  phoneNumber: string;
+  email: string;
+  sendMessageMutation: SendMessageMutation;
+};
+
+export const SubmitCodeForm: FunctionComponent<SubmitCodeFormProps> = ({ phoneNumber, email, sendMessageMutation }) => {
+  const [code, setCode] = useState('');
+
+  const {
+    mutate: submitCode,
+    data: submitCodeResponse,
+    error: submitCodeError,
+    isLoading: isLoadingSubmitCode,
+  } = useSubmitCode();
+
+  return (
+    <form
+      className={classNames(formStyles.useCaseForm, styles.codeForm)}
+      onSubmit={(e) => {
+        e.preventDefault();
+        submitCode({ code, phoneNumber });
+      }}
+    >
+      <SendMessageButton
+        sendMessageMutation={sendMessageMutation}
+        phoneNumber={phoneNumber}
+        email={email}
+        type='button'
+      />
+      <label>Verification code</label>
+      <span className={formStyles.description}>Enter the 6-digit code from the SMS message.</span>
+      <input
+        type='text'
+        name='code'
+        placeholder='Code'
+        required
+        value={code}
+        pattern='[0-9]{6}'
+        data-testid={TEST_IDS.smsFraud.codeInput}
+        onChange={(e) => setCode(e.target.value)}
+      />
+      {submitCodeError ? (
+        <Alert severity='error' className={styles.alert}>
+          {submitCodeError.message}
+        </Alert>
+      ) : null}
+      {submitCodeResponse ? (
+        <Alert severity={submitCodeResponse.severity} className={styles.alert}>
+          {submitCodeResponse.message}
+        </Alert>
+      ) : null}
```

Comment:
nit: We could just do `condition && <Component />` to make it a bit easier to read

Reply:
Fixed, thanks :) https://github.com/fingerprintjs/fingerprintjs-pro-use-cases/pull/135/commits/218fc51971d7d095ee7c50c87956bc6e8afcbcfb